### PR TITLE
Adding an ipython notebook to split the dataset for ssd7

### PR DESCRIPTION
**Pull request by gevero**
The pull request adds a notebook to facilitate the usage of the ssd7 example. Once obtained the consolidated car dataset from udacity, the notebook allows to split the `labels.csv` file in `train.csv` and `val.csv` to proceed with the training of the model.

### CODE REVIEW

**Comment by gevero** [comment]
Hi Pierluigi

Please let me know if you do not mean to merge the pull request, so that I can close it.

Best

Giovanni

**Comment by pierluigiferrari** [comment]
Hey Giovanni, I've only taken a brief look at this now. Thanks for making this, looks great! Literally a day or two before you sent me this pull request I uploaded a new version of the Udacity road traffic dataset with a train/val split due to popular demand, so the need for a split tool has somewhat decreased, but this is still useful for other datasets with annotations in CSV format.

Two comments after looking at this briefly:
1. For datasets that consist of image sequences extracted from videos (as is the case with the Udacity road traffic dataset) it would be better to set `shuffle=False` in `train_test_split()`, otherwise parts of the resulting training and validation datasets might end up being too similar, thus defeating the purpose of the validation dataset to measure the model's ability to generalize.
2. This would serve better as a function rather than loose code in a Jupyter notebook. I'm planning to unclutter the repo a bit, so I'll have to think about where this could best go before I would accept a pull request. There could be a new module with preprocessing utilities, or this could be a function in one of the existing modules, I don't know yet.

**Comment by gevero** [comment]
Hi Pierluigi

Thanks for answering me. Regarding your observations:

1.  I understand your advice, nevertheless I basically see two scenarios: 
a. We set `shuffle=False` so that the validation set serves its purpose, but in this case the model is learning from a training set filled with highly temporally correlated images, and therefore might struggle to generalize.
b. We set `shuffle=True` so that training and validation set come from the same distribution, and the convnet might a priori generalize a little better, even if we cannot tell it from the validation set. Anyway I am going with your suggestion, since my experience with sequences of images is very limited.
2. No problem in making it a function. I'll refactor it. Anyway my pull request can wait and if at some time you find it useful I will be happy to help.

One final question: is it ok to open an issue to discuss about transfer learning with ssd?

Best

Giovanni

**Comment by pierluigiferrari** [comment]
1. Unless the validation split is very large, the training images will be highly temporally correlated no matter what, so scenario (a) is not an issue. For example, I made a train/val split of roughly 80/20 for the Udacity dataset. Moving roughly every fifth image from the training dataset to the validation dataset is not going to significantly reduce the temporal dependency of the images in the training dataset. But when turning this into a function you would add a `shuffle` argument anyway, so everybody can set it how they see fit.
2. Sounds good!

> One final question: is it ok to open an issue to discuss about transfer learning with ssd?

Yeah, absolutely!
